### PR TITLE
Refactor Fetch SQL queries in Postgres backend

### PR DESCRIPTION
**Pull request by rblaine95**
An attempt at some optimizations on the `FETCH_QUERY` and `FETCH_QUERY_UPDATE` Postgres queries.

Opening as a draft PR while I work on figuring out a way to benchmark the updated queries.

According to GPT:
> The refactored versions with the LEFT JOIN and GROUP BY clauses, as well as the use of CTEs are likely to be more efficient in terms of performance, especially when dealing with large datasets. It allows for a more optimized approach to data retrieval, particularly when combining information from multiple tables.

Please be as pedantic as possible (`-Wclippy::pedantic -Wclippy::nursery`).

Related to #195

### CODE REVIEW

**Comment by andrewwhitehead** [comment]
It appears that the latest ring dependency breaks support for the manylinux aarch64 environment we are using (https://github.com/briansmith/ring/issues/1728). We could update to `manylinux_2_28` or try one of the other fixes, I'll have to test that in another PR.

**Comment by rblaine95** [comment]
Small update - I've made some progress on building a benchmark using [Criterion](https://crates.io/crates/criterion) to test if there's any benefit in the refactored SQL queries.

Preliminary results are looking good.
On my local (M2 Macbook Pro (10CPU, 16GiB)), I pre-populated the DB with 5000 profiles, then benchmarked the old SQL vs the new SQL creating 1000 profiles on top of the existing 5000.

Red = Old SQL
Blue = Refactored SQL
image

I'll work on a larger benchmark - pre-populate with 50000 profiles and then compare the SQL queries.
The expectation is that the refactored version _should_ be faster compared to the old version as the number of profiles increases.

File: askar-storage/src/backend/postgres/mod.rs [hunk 1]
```diff
@@ -49,20 +49,58 @@ const COUNT_QUERY: &str = "SELECT COUNT(*) FROM items i
...
+            AND (i.expiry IS NULL OR i.expiry > CURRENT_TIMESTAMP)
+        GROUP BY
+            i.id,
+            i.value
+    )
+    SELECT
+        *
+    FROM
+        fetched_data
+    FOR UPDATE";
```
Comment (andrewwhitehead) [inline askar-storage/src/backend/postgres/mod.rs:103]: Any reason this isn't using `FOR NO KEY UPDATE` now?

Comment (rblaine95) [reply askar-storage/src/backend/postgres/mod.rs]: Great catch, thank you!

I didn't realize I had missed that.

**Comment by rblaine95** [comment]
@swcurran, @andrewwhitehead, is there a channel on the Hyperledger Discord we could chat in?
I realized I may have been a little overly excited with my benchmark results earlier and that the tests/benchmarks aren't actually running in Multitenant mode. I'd like to figure out a way to pre-populate the DB with a number of profiles so that the benchmarking results are a little more appropriate.

**Comment by swcurran** [comment]
There is an aries-askar channel on the Hyperledger Discord Server (invite: https://discord.gg/hyperledger) that I just created that we can use.  Should have been created long ago :-).

**Pull request closed without merging**